File: v25_iterator/python/book1.py
class Book:
    def __init__(self, title):
        self.title = title

    def get_title(self):
        return self.title


class Bookshelf:
    def __init__(self):
        self.books = []

    def add_book(self, book):
        self.books.append(book)

    def get_books(self):
        return self.books


if __name__ == "__main__":
    bookshelf = Bookshelf()
    bookshelf.add_book(Book("Book 1"))
    bookshelf.add_book(Book("Book 2"))
    bookshelf.add_book(Book("Book 3"))

    books = bookshelf.get_books()
    for book in books:
        print(book.get_title())

File: v25_iterator/java/Book1.java
package v25_iterator.java;

import java.util.ArrayList;

class Book {
    private String title;

    public Book(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }
}

class Bookshelf {
    private ArrayList<Book> books;

    public Bookshelf() {
        books = new ArrayList<>();
    }

    public void addBook(Book book) {
        books.add(book);
    }

    public ArrayList<Book> getBooks() {
        return books;
    }
}

public class Book1 {
    public static void main(String[] args) {
        Bookshelf bookshelf = new Bookshelf();
        bookshelf.addBook(new Book("Book 1"));
        bookshelf.addBook(new Book("Book 2"));
        bookshelf.addBook(new Book("Book 3"));

        ArrayList<Book> books = bookshelf.getBooks();
        for (Book book : books) {
            System.out.println(book.getTitle());
        }
    }
}